src/transcoding/transcode/hwaccels/vaapi.c

```diff
@@ -37,7 +36,9 @@ typedef struct tvh_vaapi_device {
 
 
 typedef struct tvh_vaapi_context_t {
-    struct vaapi_context;
+    void *display;
```

Comment:
Why are these now in here? Where are they used?

Reply:
vaapi.h will be removed in ffmpeg5
In order to remove the dependency from vaapi_context, I copied the struct content into tvh_vaapi_device.
In my opinion the struct tvh_vaapi_device is identical before and after change (let me know if you disagree).


vaapi.h (formatting is not working ... for same reason)

`
struct attribute_deprecated vaapi_context {
    /**
     * Window system dependent data
     *
     * - encoding: unused
     * - decoding: Set by user
     */
    void *display;

    /**
     * Configuration ID
     *
     * - encoding: unused
     * - decoding: Set by user
     */
    uint32_t config_id;

    /**
     * Context ID (video decode pipeline)
     *
     * - encoding: unused
     * - decoding: Set by user
     */
    uint32_t context_id;
};
`

Reply:
No the struct is not identical. Previously it contained a struct, now it contains individual variables.

Reply:
Also I fixed the formatting of your comment so proper code tags are used

Reply:
Try just deleting it without adding any other variables. Seems like it's not used at all.

Reply:
It doesn't work because is not able to resolve TVHVAContext->config_id, TVHVAContext->context_id and TVHVAContext->display. My understanding of 'struct in a struct' is that compiler will instantiate the struct vaapi_context with his variable inside TVHVAContext. Somehow the compiler is resolving those TVHVAContext->display (aldo the display is not defined explicit in struct TVHVAContext). If I copy the variable from vaapi_context  inside TVHVAContext when I access that struct instance is going to find those 3 variable.
![error](https://user-images.githubusercontent.com/4031821/228739145-cbca00b5-f90c-44be-a175-e34ab214e0bb.png)